lib/std/json/stringify.zig

```diff
@@ -0,0 +1,606 @@
+const std = @import("std");
+const mem = std.mem;
+const assert = std.debug.assert;
+
+pub const StringifyOptions = struct {
+    pub const Whitespace = struct {
+        /// How many indentation levels deep are we?
+        indent_level: usize = 0,
+
+        /// What character(s) should be used for indentation?
+        indent: union(enum) {
+            Space: u8,
+            Tab: void,
+            None: void,
+        } = .{ .Space = 4 },
+
+        /// After a colon, should whitespace be inserted?
+        separator: bool = true,
+
+        pub fn outputIndent(
+            whitespace: @This(),
+            out_stream: anytype,
+        ) @TypeOf(out_stream).Error!void {
+            var char: u8 = undefined;
+            var n_chars: usize = undefined;
+            switch (whitespace.indent) {
+                .Space => |n_spaces| {
+                    char = ' ';
+                    n_chars = n_spaces;
+                },
+                .Tab => {
+                    char = '\t';
+                    n_chars = 1;
+                },
+                .None => return,
+            }
+            try out_stream.writeByte('\n');
+            n_chars *= whitespace.indent_level;
+            try out_stream.writeByteNTimes(char, n_chars);
+        }
+    };
+
+    /// Controls the whitespace emitted
+    whitespace: ?Whitespace = null,
```

Comment:
This looks like a good opportunity to make this non-optional since currently `null` seems to mean the same as `.{ .indent = .None, .separator = false }`.

The enums could also be made snake_case.

Reply:
Ah, I missed these enums! good catch.

This was one part of the code that i made very minor changes to, (moving things between files looks like a complete reimplementation in the diff, but we can blame git for that.) so I didn't pay too close attention to it. I might argue that it's out of scope for this PR, but since we're breaking everything anyway, it's probably a good idea to sneak in the enum change as well.

I don't know about the default value change. I can look into that.

Generally, I didn't touch the to-json part of this package much. I focused on the from-json part.